### PR TITLE
pull/push: do half the requests in case of locales#index and formats#…

### PR DESCRIPTION
**Pull request by sacry-dyn**
…index actions when pulling and pushing

### CODE REVIEW

**Comment by sacry-dyn** [comment]
@tobstarr could you review?

File: push.go [hunk 1]
```diff
@@ -35,15 +34,33 @@ func (cmd *PushCommand) Run() error {
 		return err
 	}
 
-	formats, err := client.FormatsList(1, 25)
-	if err == nil {
-		err = sources.setFormats(formats)
-		if err != nil {
+	formatMap, _ := GetFormats(client)
```
Comment (tobstarr) [inline push.go]: what is swallowed here? if it is an error, always handle it!

File: push.go [hunk 2]
```diff
@@ -35,15 +34,33 @@ func (cmd *PushCommand) Run() error {
...
+		formatName := source.GetFileFormat()
+		if val, ok := formatMap[formatName]; ok {
+			source.Format = val
+		}
+		if source.Format == nil {
+			return fmt.Errorf("Format %q of source %q is not supported by PhraseApp!", source.Format.ApiName, source.File)
 		}
 	}
 
+	projectIds := []string{}
```
Comment (tobstarr) [inline push.go]: this looks quite like a duplication of this bit here https://github.com/phrase/phraseapp-client/pull/43/files#diff-27c9dec751c416fdf10e5e5e780b008cR35

Maybe `Targets` and `Sources` could just implement an interface `ProjectIDs() []string`?

**Comment by sacry-dyn** [comment]
@tobstarr valid points, will look at them when I am back

**Comment by sacry-dyn** [comment]
@tobstarr any updates on this?

**Pull request merged**